### PR TITLE
Ft add product reviews #187584931

### PR DESCRIPTION
**Pull request by solangeihirwe03**

### CODE REVIEW

File: src/routes/productRouter.ts [hunk 1]
```diff
@@ -108,5 +109,6 @@ router.get("/buyer-view-whishlist-product",userAuthorization(["buyer"]),isUserWi
 router.get("/buyer-view-whishlist-product/:id",userAuthorization(["buyer"]),isUserWishlistExistById,productController.buyerViewWishList)
 router.delete("/delete-whishlist-products",userAuthorization(["buyer"]),isUserWishlistExist,productController.buyerDeleteAllProductFromWishlist)
 router.delete("/delete-whishlist-product/:id",userAuthorization(["buyer"]),isUserWishlistExistById,productController.buyerDeleteProductFromWishList)
+router.post("/buyer-review-product/:id",userAuthorization(["buyer"]), isProductOrdered, productController.buyerReviewProduct )
```
Comment (key-joshua) [inline src/routes/productRouter.ts]: Where is the validation for the body request of this endpoint ?

Comment (key-joshua) [inline src/routes/productRouter.ts]: What is the use of this id parameter **`(:id)`**, I am not able to see where you have used it. And it was the one you were supposed to use.

File: src/middlewares/validation.ts [hunk 2]
```diff
@@ -779,6 +779,39 @@ const isUserWishlistExistById = async (
   }
 };
 
+const isProductOrdered = async (
+  req: ExtendRequest,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const carts = await cartRepositories.getCompletedCartsByUserId(req.user.id);
```
Comment (key-joshua) [inline src/middlewares/validation.ts]: Change the logic and then start by getting a **`cartProducts`** by **`productId`** then include **`Cart`** and then check if the **`cart.status === 'completed'`** and finally check with this condition **`cart.userId !== req.user.id`**. Once everything is good proceed to the next logic.

**Review by ndahimana154 — DISMISSED** [review]

**Review by Fabrice-Dush — DISMISSED** [review]

**Review by Y-elv — DISMISSED** [review]

File: src/modules/product/controller/productController.ts [hunk 3]
```diff
@@ -387,6 +387,16 @@ const buyerViewWishList = async (req: ExtendRequest, res: Response) => {
   }
 };
 
+const buyerReviewProduct = async (req: ExtendRequest, res: Response) => {
+  const { rating, feedback } = req.body
+  const productId = req.params.id
```
Comment (key-joshua) [inline src/modules/product/controller/productController.ts]: There is no need to declare this variable as it's not re-usable.

File: src/modules/product/controller/productController.ts [hunk 4]
```diff
@@ -387,6 +387,16 @@ const buyerViewWishList = async (req: ExtendRequest, res: Response) => {
   }
 };
 
+const buyerReviewProduct = async (req: ExtendRequest, res: Response) => {
+  const { rating, feedback } = req.body
+  const productId = req.params.id
+  const createdReview = await productRepositories.userCreateReview({ rating, feedback, productId, userId: req.user.id })
+  return res.status(httpStatus.OK).json({
+    message: "Product reviewed successfully",
+    data: createdReview
```
Comment (key-joshua) [inline src/modules/product/controller/productController.ts]: Please follow the conventional code format we are using to return the response data.

File: src/modules/product/controller/productController.ts [hunk 5]
```diff
@@ -393,6 +393,20 @@ const buyerViewWishList = async (req: ExtendRequest, res: Response) => {
   }
 };
 
+const buyerReviewProduct = async (req: ExtendRequest, res: Response) => {
+  const data = {
```
Comment (key-joshua) [inline src/modules/product/controller/productController.ts]: Please use try and catch

**Pull request merged**